src/main/scala/com/twitter/finagle/postgres/PostgresClientImpl.scala

```diff
@@ -80,20 +77,25 @@ class PostgresClientImpl(
     customTypesResult
   }
 
-  private[postgres] val typeMap = types.map(Future(_)).getOrElse(retrieveTypeMap())
+
+  private[postgres] val typeMap = Refresh.every(1.hour) {
+    types.map(Future(_)).getOrElse(retrieveTypeMap())
+  }
 
   // The OIDs to be used when sending parameters
-  private[postgres] val encodeOids = typeMap.map {
-    tm => tm.toIndexedSeq.map {
-      case (oid, PostgresClient.TypeSpecifier(receiveFn, typeName, elemOid)) => typeName -> oid
-    }.groupBy(_._1).mapValues(_.map(_._2).min)
+  private[postgres] val encodeOids = Refresh.every(1.hour) {
```

Comment:
Do we need the extra `Refresh` here?  This should have the same effect:

``` scala
  private[postgres] val encodeOids = () => typeMap().map {
    tm => tm.toIndexedSeq.map {
      case (oid, PostgresClient.TypeSpecifier(receiveFn, typeName, elemOid)) => typeName -> oid
    }.groupBy(_._1).mapValues(_.map(_._2).min)
  }
```

Reply:
Thanks @dangerousben, I removed the extra Refresh.